torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -64,8 +64,13 @@ def check_node_safe(node: Node):
     Checks that the node only uses supported operators. We are starting with very
     conservative cacheability constraints, and incrementally adding more support as we expand.
     """
+    SAFE_TORCH_MODULES = ("torch.functional", "torch.nn.functional", "torch")
 
     def is_torch_function(target):
+        if isinstance(target, torch._ops.OpOverload):
+            return True
+        if hasattr(target, "__module__") and target.__module__ in SAFE_TORCH_MODULES:
```

Comment:
Just to leave it here in its own comment: I want a definition of "safe" in the docstring for `check_node_safe` so one can verify it. My understanding of the definition is:
- our cache key is computed from the FX graph produced by Dynamo and the input example values
- something is "safe" if the same cache key results in a compiled artifact that has the same behavior

Reply:
Thanks, will add! Also added some tests + an extra check that the namespace for a function under a SAFE_TORCH_MODULE is not private. If it is private, we will still allow it if it's a builtin function (i.e. implemented in C++). 

There's still a danger of someone monkeypatching a public API into torch and trying to cache that, which I left on a TODO for one of the tests. But I think that edge case is too specific for us to catch here, unless there's an easy way to distinguish between them that I haven't thought of.